x-pack/plugins/alerting/server/task_runner/task_runner.ts

```diff
@@ -466,11 +475,16 @@ export class TaskRunner<
       });
     }
 
-    let triggeredActions: AlertAction[] = [];
```

Comment:
We used to keep an array of triggered actions to calculate, numberOfTriggeredActions.
But, since the alert executions are parallel and we have to stop the executions when total numberOfTriggeredActions are exceeds the maxExecutableActions we have to keep the total number in somewhere out of the alert execution context, so they can share the number. Triggered actions array stil could've been used but, since i figured out that an execution may trigger thousands of actions didn't want to keep such a big object in the memory...

Therefore i changed the implementation a bit, rather than getting the triggered actions per alert and merge them afterwards, each execution mutates the total number in the shared store (below alertExecutionStore).